### ISSUE TITLE
[BUG] Normalize does not affect interaction features

### ISSUE THREAD

**Issue by jultou-raa** [pycaret/pycaret#1799]
**Describe the bug**

When using `normalize=True` option in regression setup coupled with other options like `feature_interaction=True`, the resulting train / test is not properly normalized.

**To Reproduce**
You can try with any dataset from sklearn. For this bug I used the diabetes one.

```python
import pandas
from sklearn.datasets import load_diabetes

data = load_diabetes(as_frame=True)
df = data.data
df['target'] = data.target
display(df.describe())
```

At this moment we have a normalized input data.

|       |           age |           sex |           bmi |            bp |            s1 |            s2 |            s3 |            s4 |           s5 |            s6 |   target |
|:------|--------------:|--------------:|--------------:|--------------:|--------------:|--------------:|--------------:|--------------:|-------------:|--------------:|---------:|
| count | 442           | 442           | 442           | 442           | 442           | 442           | 442           | 442           | 442          | 442           |  442     |
| mean  |  -3.63962e-16 |   1.30991e-16 |  -8.01395e-16 |   1.28982e-16 |  -9.04254e-17 |   1.30112e-16 |  -4.56397e-16 |   3.86317e-16 |  -3.8481e-16 |  -3.39849e-16 |  152.133 |
| std   |   0.047619    |   0.047619    |   0.047619    |   0.047619    |   0.047619    |   0.047619    |   0.047619    |   0.047619    |   0.047619   |   0.047619    |   77.093 |
| min   |  -0.107226    |  -0.0446416   |  -0.0902753   |  -0.1124      |  -0.126781    |  -0.115613    |  -0.102307    |  -0.0763945   |  -0.126097   |  -0.137767    |   25     |
| 25%   |  -0.0372993   |  -0.0446416   |  -0.0342291   |  -0.0366564   |  -0.0342478   |  -0.0303584   |  -0.0351172   |  -0.0394934   |  -0.0332488  |  -0.033179    |   87     |
| 50%   |   0.00538306  |  -0.0446416   |  -0.00728377  |  -0.00567061  |  -0.00432087  |  -0.00381907  |  -0.00658447  |  -0.00259226  |  -0.00194763 |  -0.0010777   |  140.5   |
| 75%   |   0.0380759   |   0.0506801   |   0.031248    |   0.0356438   |   0.028358    |   0.0298444   |   0.0293115   |   0.0343089   |   0.0324332  |   0.0279171   |  211.5   |
| max   |   0.110727    |   0.0506801   |   0.170555    |   0.132044    |   0.153914    |   0.198788    |   0.181179    |   0.185234    |   0.133599   |   0.135612    |  346     |

Then we setup the regression like 

```python
_ = reg.setup(
    data=df,
    target='target',
    numeric_features=df.drop(columns='target').columns.tolist(),
    normalize=True,
    silent=True,
    verbose=False,
    feature_interaction=True,
)
```
And by displaying `reg.get_config("X_train")` we have:

|       |           age |           sex |           bmi |            bp |            s1 |            s2 |         s3 |           s4 |            s5 |            s6 |   s2_multiply_bmi |   s6_multiply_bmi |   s4_multiply_bmi |
|:------|--------------:|--------------:|--------------:|--------------:|--------------:|--------------:|-----------:|-------------:|--------------:|--------------:|------------------:|------------------:|------------------:|
| count | 309           | 309           | 309           | 309           | 309           | 309           | 309        | 309          | 309           | 309           |       309         |       309         |      309          |
| mean  |  -9.25897e-09 |   2.31474e-08 |   1.54316e-09 |  -2.00611e-08 |   4.62949e-09 |   1.08021e-08 |   0        |   0          |  -1.08021e-08 |   1.62032e-08 |         0.266285  |         0.355342  |        0.41476    |
| std   |   1.00162     |   1.00162     |   1.00162     |   1.00162     |   1.00162     |   1.00162     |   1.00162  |   1.00162    |   1.00162     |   1.00162     |         0.982678  |         1.04209   |        0.981649   |
| min   |  -2.18553     |  -0.8693      |  -1.80984     |  -2.15295     |  -2.63898     |  -2.4219      |  -1.93481  |  -1.57442    |  -2.58833     |  -2.9412      |        -5.60182   |        -2.91857   |       -2.1833     |
| 25%   |  -0.774107    |  -0.8693      |  -0.71857     |  -0.68685     |  -0.715829    |  -0.602477    |  -0.724499 |  -0.789851   |  -0.67078     |  -0.696731    |        -0.164014  |        -0.158479  |       -0.00760894 |
| 50%   |   0.117318    |  -0.8693      |  -0.138108    |  -0.0904702   |  -0.0936317   |  -0.0807769   |  -0.119342 |  -0.00528125 |  -0.0712602   |   0.0215004   |         0.0967204 |         0.0807692 |        0.0573502  |
| 75%   |   0.785887    |   1.15035     |   0.651319    |   0.655004    |   0.613411    |   0.649603    |   0.561459 |   0.779288   |   0.672954    |   0.560174    |         0.555265  |         0.589326  |        0.787632   |
| max   |   2.27159     |   1.15035     |   3.66972     |   2.89143     |   3.13048     |   4.12543     |   3.66289  |   3.98818    |   2.77504     |   2.9842      |         4.95668   |         5.11823   |        5.75049    |

**Expected behavior**

As we can see the interaction features are not scaled properly. It is expected to have a 0 mean and unit standard deviation.

**Versions**

```
pycaret 2.3.4
pandas 1.3.3
scikit-learn 0.23.2
numpy 1.19.5
scipy 1.5.4
```

**Comment by moezali1** [comment]
@tvdboom Can you look into this? If it's a bug then we can fix it in our revamp.

@Yard1 Any quick fix for master? What do you think?

**Comment by tvdboom** [comment]
This is not really a bug but more a design flaw in preprocess at the moment. First, the features are normalized and after that, the new interaction features are created, which, of course, are no longer normalized. A solution would be to apply the normalization step after all feature creation steps are applied. This probably also happens with other transformation steps in preprocessing like yeo-johnson transformation. I'll make sure this is solved in the new release

**Comment by Yard1** [comment]
@moezali1 we could just change the order on master for now

**Comment by jultou-raa** [comment]
> This is not really a bug but more a design flaw in preprocess at the moment. First, the features are normalized and after that, the new interaction features are created, which, of course, are no longer normalized. A solution would be to apply the normalization step after all feature creation steps are applied. This probably also happens with other transformation steps in preprocessing like yeo-johnson transformation. I'll make sure this is solved in the new release

Ok thank you for the feedback. Can't wait to try the features selection and features interaction working properly together 😄

**Comment by moezali1** [comment]
@Yard1 Let's hold on to this on master for now.

**Comment by jultou-raa** [comment]
Is it available on pycaret-nightly package ?
